core/src/main/java/hudson/model/UpdateSite.java

```diff
@@ -1276,6 +1277,32 @@ public boolean isNeededDependenciesCompatibleWithInstalledVersion(PluginManager.
             return getDependenciesIncompatibleWithInstalledVersion(cache).isEmpty();
         }
 
+        /**
+         * Returns true if and only if this update addressed a currently active security vulnerability.
+         *
+         * @return true if and only if this update addressed a currently active security vulnerability.
+         */
+        @Restricted(NoExternalUse.class) // Jelly
+        public boolean fixesSecurityVulnerabilities() {
+            final PluginWrapper installed = getInstalled();
+            if (installed == null) {
+                return false;
+            }
+            boolean allWarningsStillApply = true;
+            for (Warning warning : ExtensionList.lookupSingleton(UpdateSiteWarningsMonitor.class).getActivePluginWarningsByPlugin().getOrDefault(installed, Collections.emptyList())) {
+                boolean thisWarningApplies = false;
+                for (WarningVersionRange range : warning.versionRanges) {
+                    if (range.includes(new VersionNumber(version))) {
+                        thisWarningApplies = true;
+                    }
+                }
+                if (!thisWarningApplies) {
+                    allWarningsStillApply = false;
+                }
+            }
+            return !allWarningsStillApply;
```

Comment:
NIT: Maybe this variable should be named `addressesActiveVuln` or something to that effect and have its values reversed in this function so that it matches the javadoc

Reply:
I also find it quite confusing. In the theory in case there is a warning for an individual plugin, then the whole message should be displayed? I think this is actually doing the opposite?